### PR TITLE
[vcpkg/scripts/pkgconfig] rewrite pkg-config check for libraries

### LINKED ISSUE #11103: [vcpkg] TODO: Use pkg-config in vcpkg_fixup_pkgconfig to check for libraries
Currently `vcpkg_fixup_pkgconfig` uses a hand written parser to get the libs used in the *.pc file but this is incomplete and does not always work as intended. As such it is probably better to directly use pkg-config calls to analyse the generated *.pc files for correctness.

### PR DESCRIPTION
**Pull request by Neumann-A**
closes #11103

- [x] Testing with X windows PR (Linux)
- [x] Testing with X windows PR (Windows) (ran until xlib and than run into other problems)

### CODE REVIEW

**Comment by JackBoosY** [comment]
Maybe we should choose one port to test this PR?

File: ports/x264/portfile.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 set(X264_VERSION 157)
 
-vcpkg_fail_port_install(ON_TARGET "OSX") 
+vcpkg_fail_port_install(ON_TARGET "OSX")
```
Comment (JackBoosY) [inline ports/x264/portfile.cmake:3]: Please update the version info. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by Neumann-A** [comment]
@JackBoosY: osx has no pkg-config installed?

**Comment by JackBoosY** [comment]
Yes, please install `brew install pkgconfig` here:
https://github.com/microsoft/vcpkg/blob/d9b4acf02d8c784927f89d23edb677408ec428af/scripts/azure-pipelines/osx/azure-pipelines.yml#L25

**Comment by Neumann-A** [comment]
@JackBoosY: Done

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Looks good in my side.

**Comment by Neumann-A** [comment]
@JackBoosY: Sorry for adding more but I found a few flaws by running the x window pr a bit further. I also added a few *.pc files to basic ports for simple testing.

**Comment by BillyONeal** [comment]
No changes to vm provisioning scripts here, dropping vm-update-request.

**Comment by Neumann-A** [comment]
Hmm osx ci is not finding pkg-config again although the brew script claims it has been installed. Don't know what to do about that. Except for th fltk regression which seems to be an error in the baseline the rest are download errors.

**Comment by JackBoosY** [comment]
Looks good now.

**Comment by JackBoosY** [comment]
Can you merge the latest changes into this branch?

**Comment by JackBoosY** [comment]
Coud you resolve these file conflicts?

**Comment by JackBoosY** [comment]
@BillyONeal This PR is important, we need to merge this first.

**Review by BillyONeal — APPROVED** [review]

**Comment by JackBoosY** [comment]
```
Building package libpng[core]:x64-osx...
...
-- Fixing pkgconfig
-- Fixing pkgconfig - release
-- Checking file: /Users/vagrant/Data/packages/libpng_x64-osx/lib/pkgconfig/libpng.pc
-- Checking file: /Users/vagrant/Data/packages/libpng_x64-osx/lib/pkgconfig/libpng16.pc
-- CHECK_LIB_m:CHECK_LIB_m-NOTFOUND
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:217 (message):
  Library "m" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:285 (vcpkg_fixup_pkgconfig_check_files)
  ports/libpng/portfile.cmake:117 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:73 (include)


Error: Building package libpng:x64-osx failed with: BUILD_FAILED
Elapsed time for package libpng:x64-osx: 8.888 s
```

Could you fix it?

Thanks.

**Comment by Neumann-A** [comment]
> Could you fix it?

Done. CI is testing the rest currently. Hopefully there is not more work to be done ;)

**Comment by Neumann-A** [comment]
@JackBoosY: Please rerun CI for libpq:x86 and libnice. The libnice failure is probably unrelated the libpq:x86 error is just a download failure for winflexbison

**Comment by Neumann-A** [comment]
@JackBoosY: needs a CI retrigger. Download errors in CI. Can this get merged any time soon?

**Comment by JackBoosY** [comment]
@ras0219  @ras0219-msft Could you merge this PR first?

**Comment by ras0219-msft** [comment]
I'm going to try to merge https://github.com/microsoft/vcpkg/pull/12254 before this PR, which will ensure all consumers are re-tested.

**Comment by ras0219-msft** [comment]
/azp run

since #12254 is merged

**Comment by Neumann-A** [comment]
currently checking why there is no regression in zstd. according to #12407 there should be a regression

**Comment by Neumann-A** [comment]
@ras0219-msft: Should have 561287c trigger a full rebuild after 5ccacd4?

Please rerun failing CIs. This PR should be green now.

**Comment by BillyONeal** [comment]
@Neumann-A Easiest way to force that is to push a merge with master; in this case there are merge conflicts

**Comment by Neumann-A** [comment]
The ompl:x86-windows regressions seems unrelated. Worked locally and error in CI is:
```
CMake Error at tests/CMakeLists.txt:16 (configure_file):
  configure_file Problem configuring file
```

**Comment by JackBoosY** [comment]
@ras0219-msft @ras0219 ping for merge this PR.

**Comment by ras0219-msft** [comment]
LGTM, thanks @Neumann-A for the continued awesome work on this!

I think that it will eventually make sense to move the "checking" parts of this into the vcpkg tool itself, to ensure it is always run on any `.pc` files left in the tree, but this is great for now.

**Pull request merged**